Source/Forms/Gen4/Wild4.cpp

```diff
@@ -465,6 +468,18 @@ void Wild4::searcherEncounterIndexChanged(int index)
             break;
         }
 
+        if ((currentProfile->getVersion() & Game::HGSS) != Game::None && (encounter == Encounter::OldRod || encounter == Encounter::GoodRod || encounter == Encounter::SuperRod))
+        {
+            if (ui->comboBoxSearcherLead->findData(toInt(Lead::SuctionCups)) < 0)
+            {
+                ui->comboBoxSearcherLead->addItem(tr("Suction Cups"), toInt(Lead::SuctionCups));
+            }
+        }
+        else if (ui->comboBoxSearcherLead->findData(toInt(Lead::SuctionCups)) >= 0)
```

Comment:
I don't think the check here is necessary. Remove doesn't do anything if the index is out of range, so we would be calling findData twice when we only need to once.

Reply:
reverted